packages/cardano-services/test/Asset/TypeormAssetProvider/TypeormAssetProvider.test.ts

```diff
@@ -0,0 +1,121 @@
+      const testAsset = assetQuery[0];
+      const asset = await provider.getAsset({ assetId: testAsset.assetId });
+      expect(asset).toEqual(testAsset);
+    });
+
+    it('should throw an error if the asset does not exist', async () => {
+      const assetId = Cardano.AssetId('64190c10b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a48656c6c6f48616e646c65');
+      await expect(provider.getAsset({ assetId })).rejects.toThrowError(ProviderError);
+    });
+
+    it('should return metadata if the asset has metadata', async () => {
+      const assetQuery = await fixtureBuilder.getAssets(1, { with: [TypeormAssetWith.metadata] });
+      const testAsset = assetQuery[0];
+      const asset = await provider.getAsset({ assetId: testAsset.assetId, extraData: { nftMetadata: true } });
+      expect(asset).toEqual(testAsset);
+    });
+
+    it('should return undefined asset token metadata if the token registry throws a server internal error', async () => {
+      const assetQuery = await fixtureBuilder.getAssets(1, { with: [TypeormAssetWith.metadata] });
+      const testAsset = assetQuery[0];
+      const asset = await provider.getAsset({ assetId: testAsset.assetId, extraData: { tokenMetadata: true } });
+      expect(asset.tokenMetadata).toBeUndefined();
+    });
+  });
+
+  describe('getAssets', () => {
+    it('should return multiple asset info for every assetId in a given array', async () => {
+      const testAssets = await fixtureBuilder.getAssets(4);
+      const assetIds = testAssets.map((asset) => asset.assetId);
+      const assets = await provider.getAssets({ assetIds });
+      expect(assets).toEqual(testAssets);
+    });
+
+    it('Should return multiple asset info with nftMetadata for every assetId in a given array', async () => {
+      const testAssets = await fixtureBuilder.getAssets(4, { with: [TypeormAssetWith.metadata] });
+      const assetIds = testAssets.map((asset) => asset.assetId);
+      const assets = await provider.getAssets({ assetIds, extraData: { nftMetadata: true } });
+      expect(assets).toEqual(testAssets);
+    });
+
+    it('Should throw error when one of the assetIds does not exist', async () => {
+      const testAssets = await fixtureBuilder.getAssets(1);
+      const invalidAssetId = Cardano.AssetId(
+        '64190c10b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a48656c6c6f48616e646c65'
+      );
+      const assetIds = [...testAssets.map((asset) => asset.assetId), invalidAssetId];
+      await expect(provider.getAssets({ assetIds })).rejects.toThrowError(ProviderError);
+    });
+
+    it('should fetch multiple token metadata for multiple assetIds', async () => {
+      const testAssets = await fixtureBuilder.getAssets(4);
+      const assetIds = testAssets.map((asset) => asset.assetId);
+      const asset = await provider.getAssets({
+        assetIds,
+        extraData: { tokenMetadata: true }
+      });
+      expect(asset).toEqual(testAssets);
```

Comment:
It seems like we don't have any tests that actually return some token metadata
```suggestion
  expect(asset).toHaveLength(testAssets);
  expect(asset[0].tokenMetadata).toBeTruthy();
```

Reply:
Fixed on 17db319